crates/mbe/src/lib.rs

```diff
@@ -369,7 +369,8 @@ pub fn expect_fragment<'t>(
 ) -> ExpandResult<tt::TokenTreesView<'t, Span>> {
     use ::parser;
     let buffer = tt_iter.remaining();
-    let parser_input = to_parser_input(edition, buffer);
+    // FIXME: Pass the correct edition per token. Due to the split between mbe and hir-expand it's complicated.
+    let parser_input = to_parser_input(buffer, &mut |_ctx| edition);
```

Comment:
Why is that? syntax context lookup isn't defined on the `ExpandDatabase` is it?

Reply:
It is.

Reply:
Ah drats, forgot we define the id in span, but not the content (we probably need to do use some extra trait setup to fix that post salsa swap to do dependency injection)